frontend/src/framework/internal/components/SaveSessionDialog/saveSessionDialog.tsx

```diff
@@ -0,0 +1,152 @@
+import React from "react";
+
+import { GuiState, useGuiState, useGuiValue } from "@framework/GuiMessageBroker";
+import { MAX_DESCRIPTION_LENGTH, MAX_TITLE_LENGTH } from "@framework/internal/persistence/constants";
+import type { Workbench } from "@framework/Workbench";
+import { Button } from "@lib/components/Button";
+import { CharLimitedInput } from "@lib/components/CharLimitedInput/charLimitedInput";
+import { CircularProgress } from "@lib/components/CircularProgress";
+import { Dialog } from "@lib/components/Dialog";
+import { truncateString } from "@lib/utils/strings";
+
+import { useActiveSession } from "../ActiveSessionBoundary";
+import { DashboardPreview } from "../DashboardPreview/dashboardPreview";
+
+export type SaveSessionDialogProps = {
+    workbench: Workbench;
+};
+
+type SaveSessionDialogInputFeedback = {
+    title?: string;
+    description?: string;
+};
+
+export function SaveSessionDialog(props: SaveSessionDialogProps): React.ReactNode {
+    const activeSession = useActiveSession();
+
+    const originalTitle = activeSession.getMetadata().title;
+    const originalDescription = activeSession.getMetadata().description ?? "";
+    const isPersisted = activeSession.getIsPersisted();
+
+    const [title, setTitle] = React.useState<string>("");
+    const [description, setDescription] = React.useState<string>("");
+
+    const [prevOriginalTitle, setPrevOriginalTitle] = React.useState<string>("");
+    const [prevOriginalDescription, setPrevOriginalDescription] = React.useState<string>("");
+
+    if (originalTitle !== prevOriginalTitle) {
+        setPrevOriginalTitle(originalTitle);
+        setTitle(`${truncateString(originalTitle, MAX_TITLE_LENGTH)}`);
+    }
+    if (originalDescription !== prevOriginalDescription) {
+        setPrevOriginalDescription(originalDescription);
+        setDescription(originalDescription);
+    }
+
+    const [isOpen, setIsOpen] = useGuiState(props.workbench.getGuiMessageBroker(), GuiState.SaveSessionDialogOpen);
+    const isSaving = useGuiValue(props.workbench.getGuiMessageBroker(), GuiState.IsSavingSession);
+    const [inputFeedback, setInputFeedback] = React.useState<SaveSessionDialogInputFeedback>({});
+    const inputRef = React.useRef<HTMLInputElement>(null);
+
+    function handleSave() {
+        if (title.trim() === "") {
+            setInputFeedback((prev) => ({ ...prev, title: "Title is required." }));
```

Comment:
Personally I'd like to see this feedback be reset when starting to type a valid title again. 

Suggestion:
- Only give feedback after save is pushed first time after dialog is opened.
- Remove feedback message when starting to type again.
- If editing text becomes empty string again, show message and red border when input field focus is lost again?